test/functional/test_framework/mininode.py

```diff
@@ -468,7 +464,7 @@ def wait_for_verack(self, timeout=60):
         def test_function():
             return self.message_count["verack"]
 
-        self.wait_until(test_function, timeout=timeout)
+        self.wait_until(test_function, timeout=timeout, check_connected=False)
```

Comment:
I think this could be improved even more by making `add_p2p_connection()` wait until `is_connected` is True before returning. That way you could leave `check_connected` as True for `wait_for_verack()` as well as in p2p_filter.py